src/state/nftMarket/reducer.ts

```diff
@@ -322,14 +321,17 @@ export const NftMarket = createSlice({
     })
     builder.addCase(fetchNftsFromCollections.fulfilled, (state, action) => {
       const { collectionAddress } = action.meta.arg
-      const existingNfts = state.data.nfts[collectionAddress] ?? []
+      const existingNfts: NftToken[] = state.data.nfts[collectionAddress] ?? []
+      const existingNftsWithoutNewOnes = existingNfts.filter(
+        (nftToken) => !action.payload.find((newToken) => newToken.tokenId === nftToken.tokenId),
+      )
 
       state.data.filters = {
         ...state.data.filters,
         loadingState: NftFilterLoadingState.IDLE,
         activeFilters: {},
       }
-      state.data.nfts[collectionAddress] = uniqBy([...existingNfts, ...action.payload], 'tokenId')
```

Comment:
@hachiojidev I think uniqBy takes the old object here, if we just want to update the list